### PR TITLE
[fluentd-gcp addon] Update Stackdriver plugin to version 0.6.7

### PR DESCRIPTION
**Pull request by crassirostris**
A new gem among all fixes Java logging severity parsing and string timestamp parsing

Also sync the buffer size with the gem guidelines, making it 1M instead of 2M.

/cc @igorpeshansky

### CODE REVIEW

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml [hunk 1]
```diff
@@ -364,6 +364,8 @@ data:
     <match kubernetes.**>
       @type google_cloud
 
+      # Try to detect JSON formatted log entries.
+      detect_json true
       # Collect metrics in Prometheus registry about plugin activity.
       enable_monitoring true
       monitoring_type prometheus
```
Comment (igorpeshansky) [inline cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Is this a good time to sync with the official config and make the buffer_chunk_size smaller (we have it at 1M)? Smaller bundles have less latency, allowing for more throughput with fewer threads... Also below.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml (hunk 1 above)
Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Done, thanks!

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml [hunk 2]
```diff
@@ -375,7 +377,7 @@ data:
       buffer_queue_full_action block
       # Set the chunk limit conservatively to avoid exceeding the GCL limit
       # of 10MiB per write request.
```
Comment (igorpeshansky) [inline cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Want to sync this [comment](https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/templates/etc/td-agent/td-agent.conf#L18) too?

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml (hunk 2 above)
Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Good catch, thanks, done

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml [hunk 3]
```diff
@@ -373,11 +375,11 @@ data:
       # Set queue_full action to block because we want to pause gracefully
       # in case of the off-the-limits load instead of throwing an exception
       buffer_queue_full_action block
-      # Set the chunk limit conservatively to avoid exceeding the GCL limit
-      # of 10MiB per write request.
-      buffer_chunk_limit 2M
+      # Set the chunk limit conservatively to avoid exceeding the Stackdriver
+      # Logging limit of 5MiB per write request.
```
Comment (igorpeshansky) [inline cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: 5MiB is not a limit, it's the recommended size. See the comment I referenced earlier (https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/templates/etc/td-agent/td-agent.conf#L18)...

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml (hunk 3 above)
Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Oh, from reading https://cloud.google.com/logging/quotas I had the impressions it is a limit

I'll fix the comment

Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml]: Thanks!

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Pull request merged**